src/index.js

```diff
@@ -67,6 +67,14 @@ MyArray.prototype.forEach = function(cb, thisArg) {
 MyArray.prototype.reduce = function(cb, initValue) {
   let accumulator = null;
```

Comment:
You don't really need this line because next your step is setting its value dependeng on initValue